kuma/core/urlresolvers.py

```diff
@@ -111,9 +111,10 @@ class Prefixer(object):
     def __init__(self, request=None, locale=None):
         """If request is omitted, fall back to a default locale."""
         self.request = request or RequestFactory(REQUEST_METHOD='bogus').request()
-        self.locale, self.shortened_path = split_path(self.request.path_info)
-        if locale:
-            self.locale = locale
+        path_locale, self.shortened_path = split_path(self.request.path_info)
+
+        # Set Self locale according to priority.
+        self.locale = locale or path_locale or self.get_chosen_language()
```

Comment:
I believe the previous version always sets ``self.locale`` to a string, possibly the empty string ``""``, while this could set it to ``None``.  I'll need to read closer to determine if this breaks the code, or you could ensure it is set to a string with another ``or ""``